static/app/components/searchQueryBuilder/formattedQuery.tsx

```diff
@@ -0,0 +1,125 @@
+import {useMemo} from 'react';
+import styled from '@emotion/styled';
+
+import {FilterValueText} from 'sentry/components/searchQueryBuilder/tokens/filter/filter';
+import {FilterKeyOperatorVisual} from 'sentry/components/searchQueryBuilder/tokens/filter/filterKeyOperator';
+import {SearchQueryBuilderParenIcon} from 'sentry/components/searchQueryBuilder/tokens/paren';
+import type {FieldDefinitionGetter} from 'sentry/components/searchQueryBuilder/types';
+import {parseQueryBuilderValue} from 'sentry/components/searchQueryBuilder/utils';
+import {
+  type ParseResultToken,
+  Token,
+  type TokenResult,
+} from 'sentry/components/searchSyntax/parser';
+import {space} from 'sentry/styles/space';
+import type {TagCollection} from 'sentry/types/group';
+import {getFieldDefinition} from 'sentry/utils/fields';
+
+export type FormattedQueryProps = {
+  query: string;
+  fieldDefinitionGetter?: FieldDefinitionGetter;
+  filterKeys?: TagCollection;
+};
+
+type TokenProps = {
+  token: ParseResultToken;
+};
+
+const EMPTY_TAGS_COLLECTION = {};
```

Comment:
needs a type otherwise is `{}`
```suggestion
const EMPTY_TAGS_COLLECTION: TagCollection = {};
```